### PR TITLE
readiness doc

### PR DESCRIPTION
**Pull request by kalantar**
Documentation for Kubernetes resource readiness checks.
Signed-off-by: Michael Kalantar <kalantar@us.ibm.com>

### CODE REVIEW

File: docs/user-guide/topics/readiness.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+## Extending to new resource kinds
+
+1. Download chart(s):
+
+    ```shell
+    iter8 hub -c load-test-http
+    ```
+
+2. Add an entry for the new resource type in `charts/iter8lib/templates/_task.ready.tpl`.
```
Comment (sriumcp) [inline docs/user-guide/topics/readiness.md:32]: I don't think we want to describe local chart editing here ... we want people to contribute to Iter8 repo instead. 
***
1. See the current implementation for deployment and service here ... 
2. As you can see, it involves two templates (`task.readiness`, and `task.ready.rbac` .. with links of course). Edit them to add support for your resource type.
3. For example, to add support for `ksvc` (Knative service), do the following ...

File: mkdocs.yml [hunk 2]
```diff
@@ -152,6 +152,7 @@ nav:
   - Topics:
     - Chart values: user-guide/topics/values.md
     - Namespace and group: user-guide/topics/group.md
+    - Resource readiness: user-guide/topics/readiness.md
```
Comment (sriumcp) [inline mkdocs.yml:155]: Could we please title this as "Readiness checks" instead?

File: docs/user-guide/topics/readiness.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
+---
+template: main.html
+---
+
+# Checking readiness of Kubernetes resources
+
+When Iter8 experiment rely on Kubernetes resources, the readiness of those resources can be checked first.
```
Comment (sriumcp) [inline docs/user-guide/topics/readiness.md:7]: Many Iter8 experiments, like `load-test-http`(with Kubernetes usage link), and `load-test-grpc` (with Kubernetes usage link) are intended to run inside a Kubernetes cluster and interact with Kubernetes resources. In such cases, it is useful to check the resources involved are *ready* at the beginning of the experiment, before other experiment tasks are executed. You can do this with various `ready.*` experiment parameters.

For example, ...

File: docs/user-guide/topics/readiness.md (hunk 1 above)
Comment (sriumcp) [reply docs/user-guide/topics/readiness.md:32]: We encourage users to contribute readiness checks for various Kubernetes resources and CRDs. ... or something like that.

**Review by sriumcp — CHANGES_REQUESTED** [review]

File: docs/tutorials/load-test-grpc/kubernetesusage.md [hunk 4]
```diff
@@ -19,19 +19,22 @@ Benchmark, and validate a gRPC service inside a Kubernetes cluster using the  [`
...
+
 Launch a `load-test-grpc` experiment inside the Kubernetes cluster. Note that the gRPC host in this experiment is `hello.default`, which refers to a hostname inside the Kubernetes cluster, specifically, the `hello` service in the `default` namespace.
+
 ```shell
 iter8 k launch -c load-test-grpc \
 --set host="hello.default:50051" \
 --set call="helloworld.Greeter.SayHello" \
---set protoURL="https://raw.githubusercontent.com/grpc/grpc-go/master/examples/helloworld/helloworld/helloworld.proto"
+--set protoURL="https://raw.githubusercontent.com/grpc/grpc-go/master/examples/helloworld/helloworld/helloworld.proto" \
+--set ready.deploy=hello \
```
[the remark this answers is not in the document]
Comment (kalantar) [reply docs/tutorials/load-test-grpc/kubernetesusage.md:36]: done

File: docs/tutorials/load-test-http/kubernetesusage.md [hunk 5]
```diff
@@ -19,25 +19,28 @@ Benchmark, and validate an HTTP service inside a Kubernetes cluster using the  [
...
-Launch a `load-test-http` experiment inside the Kubernetes cluster. Note that the HTTP URL in this experiment is `http://httpbin.default`, which refers to a hostname inside the Kubernetes cluster, specifically, the `httpbin` service in the `default` namespace. 
+
+Launch a `load-test-http` experiment inside the Kubernetes cluster. Note that the HTTP URL in this experiment is `http://httpbin.default`, which refers to a hostname inside the Kubernetes cluster, specifically, the `httpbin` service in the `default` namespace.
+
 ```shell
 iter8 k launch -c load-test-http \
 --set url=http://httpbin.default \
---set SLOs.http/latency-mean=50
+--set SLOs.http/latency-mean=50 \
+--set ready.deploy=httpbin \
```
[the remark this answers is not in the document]
Comment (kalantar) [reply docs/tutorials/load-test-http/kubernetesusage.md:34]: done

File: docs/user-guide/topics/readiness.md (hunk 1 above)
Comment (kalantar) [reply docs/user-guide/topics/readiness.md]: This is a nice idea -- encourage pull requests.
However, it is a potentially longer path to use. Local editing is much faster to get started.

Comment (kalantar) [reply docs/user-guide/topics/readiness.md]: done

File: docs/user-guide/topics/readiness.md (hunk 3 above)
Comment (kalantar) [reply docs/user-guide/topics/readiness.md]: nice. done.

**Comment by kalantar** [comment]
This has dependency on https://github.com/iter8-tools/iter8/pull/1246. There is a link that should be updated after that PR is merged.

**Review by sriumcp — APPROVED** [review]
lgtm

**Pull request merged**